### DISCUSSION TITLE
Hello, how do I get other linear solvers for ipopt?

### DISCUSSION THREAD

**Discussion by ZEROSL99** [coin-or/Ipopt#702] [Q&A] [answered]
I couldn't find the download link on the website
I'm using ipopt in windows with ampl, how do I get the .dll file to replace the linear solver,
![image](https://github.com/coin-or/Ipopt/assets/52021625/7253333a-2b30-4534-897d-c7ba288beb72)
![image](https://github.com/coin-or/Ipopt/assets/52021625/fa3d8886-1308-4ddb-8f55-5e77eea26a12)

**Comment by svigerske** [comment, marked as answer]
You can use the steps outlined at https://github.com/coin-or-tools/ThirdParty-HSL to build HSL routines from source, after obtaining them from HSL. You'll need Msys2 and a Fortran compiler, preferably Intel Fortran.

Also, https://licences.stfc.ac.uk/product/julia-hsl offers some prebuild libraries.

Btw, Ipopt 3.12.1 is pretty old.

**Comment by ZEROSL99** [comment]
Hi, I'm not quite clear about the procedure you suggested. Now I have applied to Coin-HSL for permission. I can download the following file source code, which one should I download and use with ipopt in windows environment?
![image](https://github.com/coin-or/Ipopt/assets/52021625/e0b099b1-e31b-435c-a819-df82bd60f064)

By the way, I found a libhsl.dll file in the following source package, can I use it directly?
![image](https://github.com/coin-or/Ipopt/assets/52021625/4a3db1c3-255a-4a6b-ad20-8cbae11f818f)
![image](https://github.com/coin-or/Ipopt/assets/52021625/c118e561-7f98-4100-9bac-b62f5cb5a254)

![image](https://github.com/coin-or/Ipopt/assets/52021625/f291c3d0-aa61-45f2-8a23-0c9f8d9ed23c)

**Reply by svigerske** [reply]
If you want to compile from source, then take one of the first two archives.
If you want to try the prebuild one, then take the third one (the on you seem to have downloaded already).
If the `libhsl.dll` in there works for you, then you can use it. It should have dependencies on the other DLLs in the package, so you want to copy all of them. Probably, put them next to your `ipopt.exe`.

**Comment by ZEROSL99** [comment]
Hello, what does 'prebuild one 'mean, does it mean that I can use the dll file directly?

**Reply by svigerske** [reply]
Yes, it means that someone has compiled the source already, thus 'build' a DLL from it.

**Comment by ZEROSL99** [comment]
Ok, thank you. I will try it as you suggested. By the way, I'd appreciate it if you could test it for me. Since I just started learning ampl and ipopt, there are many things I don't understand
